tests/ci/cdk/cdk/aws_lc_ec2_test_framework_ci_stack.py

```diff
@@ -84,10 +84,14 @@ def __init__(self,
                                 iam.ManagedPolicy.from_aws_managed_policy_name("AmazonSSMManagedInstanceCore"),
                                 iam.ManagedPolicy.from_aws_managed_policy_name("CloudWatchAgentServerPolicy")
                             ])
-        iam.CfnInstanceProfile(scope=self, id="{}-ec2-profile".format(id),
+        cfn_instance_prof = iam.CfnInstanceProfile(scope=self, id="{}-ec2-profile".format(id),
                                roles=["{}-ec2-role".format(id)],
                                instance_profile_name="{}-ec2-profile".format(id))
 
+        cfn_ec2_role = iam.CfnRole( self, "CfnEc2Role",
+            assume_role_policy_document=ec2_role.assume_role_policy
+        )
+        cfn_instance_prof.add_depends_on(cfn_ec2_role)
```

Comment:
Instead of explicitly adding this depends_on is there a way to pass the assume role policy to iam.CfnInstanceProfile? Also [add_depends_on](https://docs.aws.amazon.com/cdk/api/v2/python/aws_cdk.aws_iam/CfnInstanceProfile.html#aws_cdk.aws_iam.CfnInstanceProfile.add_depends_on) is deprecated.

Reply:
Fixed in latest commit